pkg/operator/azurepathfixcontroller.go

```diff
@@ -178,17 +179,10 @@ func (c *AzurePathFixController) sync() error {
 		return err
 	}
 
-	azureStorage := imageRegistryConfig.Status.Storage.Azure
-	if azureStorage == nil || len(azureStorage.AccountName) == 0 {
-		return fmt.Errorf("storage account not yet provisioned")
-	}
-	if azureStorage == nil || len(azureStorage.Container) == 0 {
-		return fmt.Errorf("storage container not yet provisioned")
-	}
-
 	// the move-blobs cmd does not work on Azure Stack Hub. Users on ASH
 	// will have to copy the blobs on their own using something like az copy.
-	if strings.EqualFold(azureStorage.CloudName, "AZURESTACKCLOUD") {
+	azureStorage := imageRegistryConfig.Status.Storage.Azure
+	if azureStorage != nil && strings.EqualFold(azureStorage.CloudName, "AZURESTACKCLOUD") {
 		return nil
 	}
 
```

Comment:
Shouldn't we still check for the presence of the `AccountName`/`Container` for the generator?

Reply:
Yeah, the generator has a [nil check](https://github.com/openshift/cluster-image-registry-operator/blob/92d64a771380eab0069fca9e4db82c784874bb4c/pkg/resource/azurepathfixjob.go?plain=1#L89-L92) but it doesn't hurt to have a more specific check here too, will do.